content/topics/Analyze/causal-inference/did/honest-did.md

```diff
@@ -0,0 +1,317 @@
+
+
+<p align = "center">
+<img src = "../images/rm-plot.png" width="500">
+</p>
+
+
+## Staggered DiD example
+
+This example demonstrates how to use the `HonestDiD` approach with a staggered DiD framework, continuing [the staggered Goodreads example](\staggered-did). 
+
+{{% tip %}}
+The full R code for this example is shared in this [GitHub Gist](https://gist.github.com/valerievossen/7eb07cadac731cfa14ce5cc44a353d4c). 
+{{% /tip %}}
+
+Follow these steps to conduct a staggered DiD sensitivity analysis:
+
+1. Specify the function
+
+To combine the staggered DiD approach of Callaway and Sant'Anna (using the `did` package) and the `HonestDiD` package, a function is created by Sant'Anna as the packages are not fully integrated yet. 
+Extract the function from the [HonestDiD GitHub page](https://github.com/asheshrambachan/HonestDiD), or from the [Gist created for this example](https://gist.github.com/valerievossen/7eb07cadac731cfa14ce5cc44a353d4c).
+
+2. Load packages and data
+
+Run the code in the [Gist](https://gist.github.com/valerievossen/7eb07cadac731cfa14ce5cc44a353d4c) to load the necessary packages and get the dataset ready to use for analysis.
+
+3. DiD estimation
+
+With the `honest_did` specified in Step 1, and the dataset `goodreads_df` loaded in Step 2, use the following for the sensitivity analyses and plots:
+
+{{% codeblock %}}
+```R
+
+# 3.1. Staggered DiD analysis 
+
+staggered_results <- did::att_gt(yname = "rating", # outcome var
+                          tname = "period_review", # time var
+                          idname = "asin", # individual var
+                          gname = "period_question", # treated
+                          control_group = "notyettreated", #control
+                          data = goodreads_df, # final dataset
+                          allow_unbalanced_panel = TRUE,
+                          base_period = "universal")
+
+# Aggregate results 
+es <- did::aggte(cs_results, type = "dynamic", na.rm = TRUE)
+
+
+# 3.2 SENSITIVITY ANALYSIS: RELATIVE MAGNITUDES 
+
+sensitivity_results_rm <-
+  honest_did(es,
+             e=0,
+             type="relative_magnitude",
+             Mbarvec=seq(from = 0.5, to = 2, by = 0.5))
+
+HonestDiD::createSensitivityPlot_relativeMagnitudes(sensitivity_results_rm$robust_ci,
+                                                    sensitivity_results_rm$orig_ci)
+
+
+# 3.3 SENSITIVITY ANALYSIS: SMOOTHNESS RESTRICTIONS
+
+sensitivity_results_sd <-
+  honest_did(es = es,
+             e = 0,
+             type="smoothness")
+
+
+HonestDiD::createSensitivityPlot(sensitivity_results_sd$robust_ci,
+                                sensitivity_results_sd$orig_ci)
+
+```
+{{% /codeblock %}}
+
+The following plots are generated:
+
+1. *Relative magnitudes*
+
+<p align = "center">
+<img src = "../images/staggered-rm-plot.png" width="500">
+</p>
+
+
+2. *Smoothness restriction*
+
+<p align = "center">
+<img src = "../images/staggered-sd-plot.png" width="500">
+</p>
+
```

Comment:
You may want to add an interpretation of these results similar to what you did for the non staggered case